src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,138 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
+import { Todo } from '../../react-app-env';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  todos: Todo[]
+  selectUser: (userId: number) => void,
+  changeTodoStatus: (todoId: number) => void,
+  selectedId: number,
+};
+
+type State = {
+  query: string,
+  selectValue: string,
+  buttonSelected: boolean,
+};
+
+export class TodoList extends React.PureComponent<Props, State> {
+  state: State = {
+    query: '',
+    selectValue: '',
+    buttonSelected: false,
+  };
+
+  handleSearchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState(() => ({
+      query: value,
+    }));
+  };
+
+  handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.currentTarget;
+
+    this.setState(() => ({
+      selectValue: value,
+    }));
+  };
+
+  selectButton = () => {
+    this.setState(state => ({
+      buttonSelected: !state.buttonSelected,
+    }));
+  };
+
+  filterTodosBySearch = () => {
+    return this.filterTodosBySelect()
+      .filter(todo => {
+        const titleLower = todo.title.toLowerCase();
+        const queryLower = this.state.query.toLowerCase();
+
+        return titleLower.includes(queryLower);
+      });
+  };
+
+  filterTodosBySelect = () => {
+    switch (this.state.selectValue) {
+      case 'All todos':
+        return this.props.todos;
+      case 'Active todos':
+        return this.props.todos.filter(todo => todo.completed === false);
+      case 'Completed todos':
```

Comment:
There are toods with two statuses - Completed and Not complete. But in your case you have active and completed.
It's confusing, let's rename this